contracts/bank_contract/v1/ergoscript/phoenix_v1_hodlcoin_bank.es

```diff
@@ -129,31 +126,26 @@
             val phoenixFeeBoxOUT: Box = OUTPUTS(2)
 
             val hodlCoinsBurned: Long = hodlCoinsOut - hodlCoinsIn
-            //val expectedAmountBeforeFees: Long = (hodlCoinsBurned * price) / precisionFactor
-            val expectedAmountBeforeFeesPrime: Long = (hodlCoinsBurned * price)
-            //val bankFeeAmount: Long = (expectedAmountBeforeFees * bankFeeNum) / feeDenom
-            val bankFeeAmountPrime: Long = (expectedAmountBeforeFeesPrime * bankFeeNum)
-            //val devFeeAmount: Long = (expectedAmountBeforeFees * devFeeNum) / feeDenom
-            val devFeeAmountPrime: Long = (expectedAmountBeforeFeesPrime * devFeeNum)
-            //val expectedAmountWithdrawn: Long = expectedAmountBeforeFees - bankFeeAmount - devFeeAmount // The buyer never gets the bankFeeAmount since it remains in the bank box, the devFeeAmount is the only ERG that leaves the box.
-            val expectedAmountWithdrawnPrime: Long = expectedAmountBeforeFeesPrime * feeDenom - (bankFeeAmountPrime - devFeeAmountPrime) * precisionFactor
-
-            //val validBankWithdraw: Boolean = (reserveOut == reserveIn - (expectedAmountBeforeFees - devFeeAmount))
-            val validBankWithdraw: Boolean = (reserveOut * (precisionFactor * feeDenom) == reserveIn * (precisionFactor * feeDenom) - (expectedAmountBeforeFeesPrime * feeDenom - devFeeAmountPrime * precisionFactor))
+            val expectedAmountBeforeFees: Long = (hodlCoinsBurned * price) / precisionFactor
+            val bankFeeAmount: Long = (expectedAmountBeforeFees * bankFeeNum) / feeDenom
+            val devFeeAmount: Long = (expectedAmountBeforeFees * devFeeNum) / feeDenom
+            val expectedUserAmount: Long = expectedAmountBeforeFees - bankFeeAmount - devFeeAmount // The buyer never gets the bankFeeAmount since it remains in the bank box.
+            
+            val validBankWithdraw: Boolean = (reserveOut == reserveIn - expectedUserAmount + bankFeeAmount) // What should have left the bank is the amount the user got plus the bank fee amount which must remain in the bank.
```

Comment:
Updated bank withdraw amount since before it did not enforce the bank fee amount remains in the bank.